### PR TITLE
Search card refresh

### PR DESCRIPTION
**Pull request by hero101**
- The search cards version does not represent 1:1 the mockups because of dimension constraints.
I tried matching the styling and structure of the mockup.
- I have also provided two class diagrams so the architecture visibility is improved and further extension of the cards is easier.

**I suggest only critical or minor issues be addressed in this PR, and bigger changes be addressed later.**

### CODE REVIEW

File: src/pages/Search/SearchPage.tsx [hunk 1]
```diff
@@ -40,7 +47,7 @@ const contributorFilterConfig: FilterConfig = {
 const entityFilterConfig: FilterConfig = {
```
Comment (ccanos) [inline src/pages/Search/SearchPage.tsx:47]: `journeyFilterConfig`

File: src/common/themes/default/defaults.ts [hunk 2]
```diff
@@ -26,6 +26,17 @@ export const theme: ThemeOptions = {
     },
   },
   cards: {
+    search: {
+      width: 264,
```
Comment (ccanos) [inline src/common/themes/default/defaults.ts:30]: The constants I'm adding here are normally passed through `theme.spacing`, for example in simpleCard down there the width is 32 * spacing(8px) = 256px. 
I see that your height for example is not a multiple of 8 which is a problem to do what I'm saying, so I don't know what I'd do...

- If it's okay to make cards a few pixels bigger or smaller to match the 8px spacing grid, great.
- If graphically they are pixel perfect, I've calculated 394 = 49.25, so contributor.height would be 49.25, that's not a terribly ugly code...
- And if you don't agree, just leave it like this, it is fine.

File: src/domain/shared/components/search-cards/base/SearchBaseCard.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import CardContent from '@mui/material/CardContent';
+import CardActions from '@mui/material/CardActions';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import TagsComponent from '../../TagsComponent/TagsComponent';
+import { ClampedTypography } from '../../ClampedTypography';
+import { RouterLink } from '../../../../../common/components/core/RouterLink';
+import Link from '@mui/material/Link';
+
+const MATCH_TERMS_HEIGHT = 50;
```
Comment (ccanos) [inline src/domain/shared/components/search-cards/base/SearchBaseCard.tsx:14]: maybe these could go in the theme constants file too

File: src/pages/Search/SearchResultCardChooser.tsx [hunk 4]
```diff
@@ -1,42 +1,278 @@
...
-  if (!card) {
+  if (card === undefined) {
     throw new Error(`Unrecognized result typename: ${result.__typename}`);
   }
 
   return card;
 };
 export default SearchResultCardChooser;
+
+const useHydrateCard = (
```
Comment (ccanos) [inline src/pages/Search/SearchResultCardChooser.tsx:74]: I'd put this in a separate file.

**Review by ccanos — APPROVED** [review]
Just minor comments

File: src/common/themes/default/defaults.ts (hunk 2 above)
Comment (hero101) [reply src/common/themes/default/defaults.ts:30]: Using the spacing requires the whole application to be based on this spacing constant.
The contents of the cards should be then dynamically calculated - lines truncated, wrapped, etc.
Having hardcoded dimensions is more reliable at this point imo.

File: src/domain/shared/components/search-cards/base/SearchBaseCard.tsx (hunk 3 above)
Comment (hero101) [reply src/domain/shared/components/search-cards/base/SearchBaseCard.tsx:14]: I really don't know. These are constants are related to the base component itself. Which will not be used directly.

**Review by ccanos — APPROVED** [review]

**Pull request merged**